src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -183,31 +171,30 @@ public void generateChecksumFiles(IncrementalTaskInputs inputs) throws IOExcepti
         }
 
         Algorithm algo = algorithm.get();
-        inputs.outOfDate(inputFileDetails -> {
-            File input = inputFileDetails.getFile();
-            if (input.isDirectory()) {
-                return;
-            }
-            File sumFile = outputFileFor(outputDirAsFile, input, algo);
-            try {
-                HashCode hashCode = Files.asByteSource(input).hash(algo.hashFunction);
-                Files.write(hashCode.toString().getBytes(), sumFile);
-            } catch (IOException e) {
-                throw new GradleException("Trouble creating checksum", e);
-            }
-        });
 
-        inputs.removed(inputFileDetails -> {
-            File input = inputFileDetails.getFile();
-            if (input.isDirectory()) {
-                return;
-            }
-            if (isFileSystemOperationsSupported()) {
-                fileOps.getFs().delete(spec ->
-                    spec.delete(outputFileFor(outputDirAsFile, input, algo))
-                );
-            } else {
-                getProject().delete(outputFileFor(outputDirAsFile, input, algo));
+        inputChanges.getFileChanges(files).forEach(change -> {
+            if (change.getFileType() == FileType.DIRECTORY) return;
+
+            switch(change.getChangeType()){
+                case ADDED:
+                case MODIFIED:
+                    File sumFile = outputFileFor(outputDirAsFile, change.getFile(), algo);
+                    try {
+                        HashCode hashCode = Files.asByteSource(change.getFile()).hash(algo.hashFunction);
+                        Files.write(hashCode.toString().getBytes(), sumFile);
+                    } catch (IOException e) {
+                        throw new GradleException("Trouble creating checksum", e);
```

Comment:
@jprinet I would write `Error` instead of `Trouble`

Reply:
And I would not throw a GradlException but a built-in JDK exception type. Gradle will wrap it anyway.

Reply:
I'm keen to any change but this was like that from the very first version.
Changing the error message or the exception type could have some impact on existing consumers.